bokeh/models/expressions.py

```diff
@@ -46,6 +46,34 @@ class Expression(Model):
     '''
     pass
 
+class CumSum(Expression):
+    ''' An expression for generating arrays by cumulatively summing a single
+    column from a ``ColumnDataSource``.
+
+    '''
+
+    field = String(help="""
+    The name of a ColumnDataSource column to cumulatively sum for new values.
+    """)
+
+    include_zero = Bool(default=False, help="""
+    Whether to include zero at the start of the result. Note that the length
+    of the result is always the same as the input column. Therefore if this
+    property is True, then the last value of the column will not be included
+    in the sum.
+
+    .. code-block:: python
+
+        source = ColumnDataSource(data=dict(foo=[1, 2, 3, 4]))
+
+        CumSum('foo')
```

Comment:
Do ``Expressions`` actually allow positional arguments? Might be the case but maybe you confused it with the cumsum helper function signature?

Reply:
No, this code is incorrect.

Reply:
Also, it would be good to use actual Python shell syntax (`>>>`).

Reply:
fixed in https://github.com/bokeh/bokeh/pull/7961/commits/7f2a0dc151138aec7add74bebf5249a1d40740bb